tests/launchsecurity/sev.go

```diff
@@ -1,27 +1,180 @@
+			APIMinor:  17,
+			BuildID:   46,
+			Policy:    1,
+			LoaderSHA: "a9da39a73edf4a7ba84e03a7e373c147c2d3df255051a6a130df4c2dd0d569d8",
+		}
+	)
+
+	newSEVFedora := func(opts ...libvmi.Option) *v1.VirtualMachineInstance {
+		const secureBoot = false
+		sevOptions := []libvmi.Option{
+			libvmi.WithUefi(secureBoot),
+			libvmi.WithSEV(),
+		}
+		opts = append(sevOptions, opts...)
+		return libvmi.NewFedora(opts...)
+	}
+
+	// As per section 6.5 LAUNCH_MEASURE of the AMD SEV specification the launch
+	// measurement is calculated as:
+	//   HMAC(0x04 || API_MAJOR || API_MINOR || BUILD ||
+	//        GCTX.POLICY || GCTX.LD || MNONCE; GCTX.TIK)
+	// The implementation is based on
+	//   https://blog.hansenpartnership.com/wp-uploads/2020/12/sevsecret.txt
+	verifyMeasurement := func(sevMeasurementInfo *v1.SEVMeasurementInfo, tikBase64 string) []byte {
+		By("Verifying launch measurement")
+		tik, err := base64.StdEncoding.DecodeString(tikBase64)
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+		h := hmac.New(sha256.New, tik)
+		binary.Write(h, binary.LittleEndian, uint8(0x04))
+		binary.Write(h, binary.LittleEndian, uint8(sevMeasurementInfo.APIMajor))
+		binary.Write(h, binary.LittleEndian, uint8(sevMeasurementInfo.APIMinor))
+		binary.Write(h, binary.LittleEndian, uint8(sevMeasurementInfo.BuildID))
+		binary.Write(h, binary.LittleEndian, uint32(sevMeasurementInfo.Policy))
+		loaderSHA, err := hex.DecodeString(sevMeasurementInfo.LoaderSHA)
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+		h.Write(loaderSHA)
+		m, err := base64.StdEncoding.DecodeString(sevMeasurementInfo.Measurement)
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+		nonce := m[32:48]
+		h.Write(nonce)
+		measure := m[0:32]
+		ExpectWithOffset(1, hex.EncodeToString(h.Sum(nil))).To(Equal(hex.EncodeToString(measure)))
+		return measure
+	}
+
+	// The implementation is based on
+	//   https://blog.hansenpartnership.com/wp-uploads/2020/12/sevsecret.txt
+	encryptSecret := func(diskSecret string, measure []byte, tikBase64, tekBase64 string) *v1.SEVSecretOptions {
+		By("Encrypting launch secret")
+		tik, err := base64.StdEncoding.DecodeString(tikBase64)
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+		tek, err := base64.StdEncoding.DecodeString(tekBase64)
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+		// AMD SEV specification, section 4.6 Endianness: all integral values
+		// passed between the firmware and the CPU driver are little-endian
+		// formatted. Applies to UUIDs as well.
+		writeUUID := func(w io.Writer, uuid uuid.UUID) {
+			binary.Write(w, binary.LittleEndian, binary.BigEndian.Uint32(uuid[0:4]))
+			binary.Write(w, binary.LittleEndian, binary.BigEndian.Uint16(uuid[4:6]))
+			binary.Write(w, binary.LittleEndian, binary.BigEndian.Uint16(uuid[6:8]))
+			w.Write(uuid[8:])
+		}
+
+		// total length of table: header plus one entry with trailing \0
+		l := 16 + 4 + 16 + 4 + len(diskSecret) + 1
+		// SEV-ES requires rounding to 16
+		l = (l + 15) & ^15
```

Comment:
This part is cryptic to me :)
Maybe it's better to introduce consts to explain these numbers' meaning?